### ISSUE TITLE
[regression in 1.70] [clippy] redundant clone ; cloned value is neither consumed nor mutated

### ISSUE THREAD

**Issue by glandium** [rust-lang/rust-clippy#10870]

**Comment by jsdw** [comment]
I ran into a similar false positive; this snippet causes the clippy warning on the rust playground:

```rust
fn main() {
    #[derive(Clone, Debug, PartialEq)]
    struct Foo { n: usize }
    
    let mut foo = Foo { n: 1 };
    let foo2 = foo.clone();
    
    foo.n = 2;
    assert_eq!(foo, foo2);
}
```

The code could be rejigged of course, but the clone here is clearly necessary since the _other_ value is being mutated and we want to keep a copy of the original.

**Comment by Alexendoo** [comment]
See also https://github.com/rust-lang/rust/pull/108944#issuecomment-1564451009
